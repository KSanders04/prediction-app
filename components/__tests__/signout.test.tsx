import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Signout from "@/components/signout";
import { Alert } from "react-native";

// Mock dependencies
jest.mock("expo-router", () => ({
  router: { replace: jest.fn() }
}));
jest.mock("@/components/firebaseFunctions", () => ({
  signOutUser: jest.fn(),
  listenForSignOut: jest.fn()
}));
jest.mock('@expo/vector-icons/FontAwesome', () => 'FontAwesome');

describe("SignOut Screen", () => {
  it("renders sign out button", () => {
    const { getByText } = render(<Signout />);
    expect(getByText("Sign Out")).toBeTruthy();
  });

  it("shows alert when sign out button is pressed", () => {
    const alertSpy = jest.spyOn(Alert, "alert").mockImplementation(() => {});
    const { getByText } = render(<Signout />);
    fireEvent.press(getByText("Sign Out"));
    // Since Alert.alert is used, you can check if it was called
    expect(alertSpy).toHaveBeenCalled();
    alertSpy.mockRestore();
  });

      it('matches snapshot', () => {
        const tree = render(<Signout />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});