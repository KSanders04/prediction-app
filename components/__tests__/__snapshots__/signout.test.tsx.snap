// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut Screen matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#3498db",
      "borderRadius": 15,
      "elevation": 5,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginTop": 10,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#5C6BC0",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 5,
      "width": "90%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <FontAwesome
      color="#FFFFFF"
      name="sign-out"
      size={25}
      style={
        {
          "marginRight": 8,
          "opacity": 1,
        }
      }
    />
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      Sign Out
    </Text>
  </View>
</View>
`;
